colossalai/shardformer/layer/_operation.py

```diff
@@ -611,6 +638,13 @@ def _reduce_scatter(input_, dim=1, process_group=None):
     return output
 
 
+def _all_to_all(input_, world_size, group, scatter_dim, gather_dim):
+    input_list = [t.contiguous() for t in torch.tensor_split(input_, world_size, scatter_dim)]
+    output_list = [torch.empty_like(input_list[0]) for _ in range(world_size)]
+    dist.all_to_all(output_list, input_list, group=group)
```

Comment:
you can use all_to_all_single

Reply:
having addressed this, using all_to_all_single when batch size is 1